instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -219,7 +219,7 @@ def instrument_app(
                 tracer_provider=tracer_provider,
                 meter=meter,
             )
-            app.is_instrumented_by_opentelemetry = True
+            app._is_instrumented_by_opentelemetry = True
```

Comment:
This is actually a bugfix, right? Can you add a test case? This could be the main change of this PR.

Reply:
I have added some assertions to the `test_metric_for_uninstrument_app_method` test. See below.